stage0/src/sev.rs

```diff
@@ -205,12 +205,32 @@ pub fn share_page(page: Page<Size4KiB>) {
 
 /// Stops sharing a single 4KiB page with the hypervisor when running with AMD SEV-SNP enabled.
 pub fn unshare_page(page: Page<Size4KiB>) {
+    let page_start = page.start_address().as_u64();
+    // Only the first 2MiB is mapped as 4KiB pages, so make sure we fall in that range.
+    assert!(page_start < Size2MiB::SIZE);
     if sev_status().contains(SevStatus::SNP_ACTIVE) {
         let page_start = page.start_address().as_u64();
         let request = SnpPageStateChangeRequest::new(page_start as usize, PageAssignment::Private)
             .expect("invalid address for page location");
         change_snp_page_state(request).expect("couldn't change SNP state for page");
     }
+    // Mark the page as encrypted.
+    {
+        let mut page_tables = crate::paging::PAGE_TABLE_REFS.get().unwrap().lock();
+        let pt = &mut page_tables.pt_0;
+        let idx = (page_start / Size4KiB::SIZE) as usize;
```

Comment:
`Page` should have a family of methods for computing these indexes: https://docs.rs/x86_64/latest/x86_64/structures/paging/page/struct.Page.html#method.page_table_index